pkg/api/validation/events.go

```diff
@@ -25,13 +29,35 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	// TODO: There is no namespace required for node.
-	if event.InvolvedObject.Kind != "Node" &&
-		event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
+
+	nonNamespacedKindFlag, err := isNonNamespacedKind(event.InvolvedObject.Kind, event.InvolvedObject.APIVersion)
+	if err != nil {
+		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.Kind", event.InvolvedObject.Kind, fmt.Sprintf("couldn't check whether namespace is allowed: %s", err)))
```

Comment:
OK, so this would start rejecting those events. I guess we can't start validating these unless we're prepared to ask OpenShift to start registering all types as 3rd party objects or something. Not even sure if that would work at the moment. @brendandburns does registering as a 3rd party object cause the rest mapper to be populated?